molgenis/molgenis-emx2
# feat: fix ontology tree ui performance and enabled large disease ontologies filter on catalogue

closes https://github.com/molgenis/GCC/issues/818
also in parts addresses https://github.com/molgenis/GCC/issues/819 (duplicate PR)

how to test:
- go to http://localhost:3000/catalogue/ssr-catalogue/testNetwork1/collections?page=1 and open 'Disease' and chose 'VII'
- see that it loads relatively quickly as compared to previous version of the PR
- look at other ontologies, they are more snappy
- also look at https://preview-emx2-pr-4539.dev.molgenis.org/apps/tailwind-components/#/input/Tree.story

N.B. with https://github.com/molgenis/molgenis-emx2/pull/4584 we should be able to use the emitSelectedChildren=false. Should probably enable it still in this PR, but added the feature to make it possible so I can continue in the other PR

todo:
- [x] added population.mainMedicalCondition filter on resources list view
- [x] addressed performance issue with showing large ontologies by changing Tree component
- [x] moved search from Ontology to Tree to also address performance issue (and drop the modal, I don't like it)
- [x] changed ontology options search from modal to inline 

N.B. this is confusing, there can also be populationDisease. We would need to produce filter on both with an 'or'? Or shall we reduce the data model so there is only one place where this is defined? I propose on the subpopulations.
